zio/zjsonio/stream.go

```diff
@@ -51,7 +52,7 @@ func encodeContainer(typ zng.Type, val []byte) (interface{}, error) {
 	}
 	childType, columns := zng.ContainedType(typ)
 	if childType == nil && columns == nil {
-		return nil, zbuf.ErrSyntax
+		return nil, ErrInvalid
```

Comment:
Since this is the only use of ErrInvalid, I think you should get rid of it and just say `return nil, errors.New("invalid container")` here.